src/precice/impl/Participant.cpp

```diff
@@ -250,124 +217,56 @@ std::vector<MeshContext *> &Participant::usedMeshContexts()
-{
-  auto pos = std::find_if(_usedMeshContexts.begin(), _usedMeshContexts.end(),
-                          [meshID](MeshContext const *context) {
-                            return context->mesh->getID() == meshID;
-                          });
-  PRECICE_ASSERT(pos != _usedMeshContexts.end());
-  return **pos;
-}
-
-MeshContext const &Participant::usedMeshContext(MeshID meshID) const
-{
-  auto pos = std::find_if(_usedMeshContexts.begin(), _usedMeshContexts.end(),
-                          [meshID](MeshContext const *context) {
-                            return context->mesh->getID() == meshID;
-                          });
-  PRECICE_ASSERT(pos != _usedMeshContexts.end());
-  return **pos;
-}
-
-MeshContext &Participant::usedMeshContext(const std::string &name)
+MeshContext &Participant::usedMeshContext(std::string_view mesh)
 {
   auto pos = std::find_if(_usedMeshContexts.begin(), _usedMeshContexts.end(),
-                          [&name](MeshContext const *context) {
-                            return context->mesh->getName() == name;
+                          [mesh](MeshContext const *context) {
+                            return context->mesh->getName() == mesh;
                           });
   PRECICE_ASSERT(pos != _usedMeshContexts.end());
   return **pos;
 }
 
-MeshContext const &Participant::usedMeshContext(const std::string &name) const
+MeshContext const &Participant::usedMeshContext(std::string_view mesh) const
 {
   auto pos = std::find_if(_usedMeshContexts.begin(), _usedMeshContexts.end(),
-                          [&name](MeshContext const *context) {
-                            return context->mesh->getName() == name;
+                          [mesh](MeshContext const *context) {
+                            return context->mesh->getName() == mesh;
                           });
   PRECICE_ASSERT(pos != _usedMeshContexts.end());
   return **pos;
 }
 
-bool Participant::hasMesh(MeshID meshID) const
+bool Participant::hasMesh(std::string_view mesh) const
 {
-  return meshID < static_cast<int>(_meshContexts.size()) && _meshContexts.at(meshID) != nullptr;
+  return _meshContexts.count(mesh) > 0;
 }
 
-bool Participant::hasMesh(const std::string &meshName) const
-{
-  return std::any_of(
-      _meshContexts.begin(), _meshContexts.end(),
-      [&meshName](const MeshContext *mcptr) {
-        return mcptr && meshName == mcptr->mesh->getName();
-      });
-}
-
-bool Participant::isMeshUsed(MeshID meshID) const
+bool Participant::isMeshUsed(std::string_view mesh) const
 {
   return std::any_of(
       _usedMeshContexts.begin(), _usedMeshContexts.end(),
-      [meshID](const MeshContext *mcptr) {
-        return mcptr->mesh->getID() == meshID;
+      [mesh](const MeshContext *mcptr) {
+        return mcptr->mesh->getName() == mesh;
       });
 }
 
-bool Participant::isMeshUsed(const std::string &meshName) const
+bool Participant::isMeshProvided(std::string_view mesh) const
```

Comment:
Why do we change the name of the function here? The two names mean two different things.

Reply:
I removed many variants, so this seems to be mainly an artefact in the diff.